# VIE-Doors Integrator Loading

In [1]:
from viedoors import CADLoader, BSTLoader, FLTLoader, HMLoader, FMLoader, NPALoader

## HM Liste (done 🆗)

```python
hm_data["GESCH."] = hm_data["GESCH."].astype(str).map(level_mapper)
hm_data["aks"] = hm_data["TÜRNR(AKS)"]
```

In [2]:
h = HMLoader(file="data/Schrack_HM_Obj420.xls", title="HM")
h.get_data(prefixed=True)

HM___objekt HM___geschloss  HM___tuer_nr_aks HM___tuer_nr_alt  \
0          420             EG   420 EG0A0106.02              004   
2          420             EG   420 EG0A0803.02             040D   
3          420             EG   420 EG0A0602.01             040A   
4          420             EG   420 EG0A0406.01              008   
5          420             EG   420 EG A0807.02              009   
7          420             EG   420 EG0A2207.01              017   
9          420             EG   420 EG0A1206.01              018   
11         420             EG   420 EG0A1308.01             011A   
12         420             EG   420 EG0A1308.02             037A   
14         nan            NaN               nan              NaN   
16         420             EG   420 EG0A3207.01              023   
17         420             EG   420 EG0A2601.02             047A   
19         420             EG   420 EG0A3106.01              022   
21         420             EG   420 EG0A3906.01              028   
23         420             EG   420 EG0A3408.01             049A   
25         420             EG   420 EG0A4001.02             062A   
27         420             EG   420 EG0A4401.07              030   
28         420             EG   420 EG0A5305.01              032   
29         420             EG    420 00A4408.01             055A   
31         420             EG   420 EG0A4001.01             064D   
33         420             EG   420 EG0A4608.06             057B   
34         420             EG   420 EG0A5206.02             032A   
36         420             EG   420 EG0A2114.01             075A   
37         420             EG   420 EG0A2114.01             075F   
39         420           1.OG   420 01A 0414.02             103B   
41         420           2.OG  420 020A 0714.01              212   
43         420           2.OG  420 020A 0416.02              216   
44         420           2.OG  420 020A 0416.01              217   
46         420             DG   420 DAA 0714.01            keien   
48         420             EG   420 EG0B2605.01            0102A   
50         420             EG   420 EG0B1607.03             0101   
52         420             EG   420 EG0B1202.02             090A   
53         420             EG   420 EG0B1302.01              086   
55         420           1.OG    420 01A0230.01             116A   
56         420           1.OG    420 01A0222.02             110A   
58         420             EG   420 EGB11605.02            0100C   
59         420             EG   420 EG0B1607.01              093   
61         420             EG   420 EG0B1005.01             083B   
62         420             EG   420 EG0B0805.01             082B   
63         420             EG   420 EG0B0405.01             081B   
65         420             EG   keine vorhanden  keine vorhanden   

   HM___bs_tuere_alt HM___raum_nr HM___aks_raum HM___uz_6_steu HM___kreis  \
0          420/0/3/6          001        0A0106              5          1   
2         420/0/14/6         040D        0A0803              6          2   
3         420/0/15/6         040A        0A0602              6          2   
4         420/0/22/6          004        0A0406              6          2   
5           420/0/10          009        0A0807              6          2   
7         420/0/40/6          017        0A2207              7          3   
9         420/0/45/6          008        0A1206              8          4   
11        420/0/29/6         011A        0A1308              9          5   
12        420/0/30/6          013        0A1607              9          5   
14               nan          NaN           NaN            NaN          6   
16        420/0/57/6          023        0A3207             12          8   
17        420/0/58/6          018        0A2601             12          8   
19        420/0/56/6          022        0A3106             13         10   
21        420/0/61/6          028        0A3906             15      

## CAD File (DONE 🆗)

* Die Spalten B bis F ergeben die AKS-Nummer (``GAR-TURNUMMER-BAUTEIL``, ``GAR-TURNUMMER-EBENE``, ``GAR-TURNUMMER-MODUL``, ``GAR-TURNUMMER-AKS-NR``, ``GAR-TURNUMMER-NUMMER``)
* Fluchttürnummer in Spalte H (``GAR-FLUCHTTUERNUMMER``)

In [4]:
c = CADLoader(file="data/420_gesamt_20250122.xlsx", title="CAD")
c.get_data(prefixed=True)

CAD___gar_tuernummer_bauteil CAD___gar_tuernummer_ebene  \
0                            420                         00   
1                            420                         00   
2                            420                         00   
3                            420                         00   
4                            420                         00   
..                           ...                        ...   
419                          420                         K1   
420                          420                         K1   
421                          420                         K1   
422                          420                         K1   
423                          420                         Z1   

    CAD___gar_tuernummer_modul CAD___gar_tuernummer_aks_nr  \
0                            A                        0104   
1                            A                        0106   
2                            A                        0106   
3                            A                        0108   
4                            A                        0108   
..                         ...                         ...   
419                          A                        2108   
420                          A                        2205   
421                          A                        2308   
422                          A                        2506   
423                          A                        4410   

    CAD___gar_tuernummer_nummer CAD___gar_tuernummer_alt  \
0                            01                     036D   
1                            01                      001   
2                            02                      004   
3                            01                     029A   
4                            02                     003A   
..                          ...                      ...   
419                          01                     5K/1   
420                          01                     5K/2   
421                          01                      01A   
422                          01                     5K/3   
423                          01                     055C   

    CAD___gar_flucht_tuer_nr CAD___gar_tuer_breite CAD___gar_tuer_hoehe  \
0                         --                    90                  200   
1                         65                   170                  200   
2                         63                   157                  200   
3                         --                   165                  200   
4                         --                    85                  200   
..                       ...                   ...                  ...   
419                       --                    80                  200   
420                       --                    80                  200   
421                       --                   112                  243   
422                       --                    80                  200   
423                       NV                    90                  ---   

    CAD___gar_bsk CAD___gar_brandabschnitt_bildend CAD___gar_hel  \
0             ---                             NEIN          NEIN   
1             ---                             NEIN          Nein   
2             T30                             NEIN          Nein   
3             T30                             NEIN          NEIN   
4             ---                             NEIN          NEIN   
..            ...                              ...           ...   
419          Nein                             NEIN           ---   
420          Nein                             NEIN           ---   
421          Nein                             NEIN           ---   
422          Nein                             NEIN           ---   
423           ---                             NEIN           ---   

    CAD___gar_sec CAD___gar_zusatz CAD___integration_aks  
0   

## NPA Liste (DONE 🆗)

Bei der Liste „Durch Mitarbeiter aufgenommene Anlagenliste (Dateinamen „NPA_Tür Aufnahme Obj. 420“)“ ist der Feuerwiderstand jenes Element, welches als Grundlage für den Vergleich mit den anderen Listen herangezogen werden soll. Folgende relevanten Elemente müssen mit den anderen Listen verglichen und aufgelistet werden

* AKS-Nummer Plan (setzt sich zusammen aus Objekt/Ebene/Bauteil/Aks Plan) – Bsp. 103 00 A0220.01
* AKS-Nummer real (setzt sich zusammen aus Objekt/Ebene/Bauteil/Aks real) – Bsp. 103 00 0A0220.01
* Schlossernummer (Bsp. 103/0/1)
* Alte Türnummer (Bsp. BW102)
* Fluchttürnummer (Spalte Q) neu

In [5]:
n = NPALoader(file="data/NPA_Tuer Aufnahme Obj420.xlsx", title="NPA")
n.get_data()

objekt ebene bauteil aks_plan   aks_real alte_tuernummer   \
0      420    K1       A  2205.01  keine Nr.              5K2   
1      420    K1       A  2506.01  keine Nr.              5K3   
2      420    K1       A  2308.01  keine Nr.              01A   
3      420    K1       A  2108.01  keine Nr.              5K1   
4      420    K1       A  1305.02  keine Nr.              3K1   
..     ...   ...     ...      ...        ...              ...   
109    420    02       A  0226.02  Keine Nr.             025B   
110    420    DA       A  0216.01    0216.01        Keine Nr.   
111    420    DA       A  0213.02    0213.02        Keine Nr.   
112    420    DA       A  0213.01    0213.01        Keine Nr.   
113    420    DA       A  0714.01  keine Nr.        Keine Nr.   

     schlossernummer hoehe_cm breite_cm zylinder__Z_B_WC_S  ...  \
0       063 / K / 48      200        80               nein  ...   
1       063 / K / 47      200        80               nein  ...   
2          keine Nr.      243       112               nein  ...   
3      063 / K / 109      200        80               nein  ...   
4      063 / K / 149      200        80               nein  ...   
..               ...      ...       ...                ...  ...   
109  Nicht vorhanden        0         0                  B  ...   
110         420/DG/1      200     90/40                  Z  ...   
111        keine Nr.      200        80                  Z  ...   
112        keine Nr.      200        80                  Z  ...   
113         420/DG/2      220       145               nein  ...   

        panik__d_stk nottaster__ja_nein offenhalte-magnet__ja_nein  \
0              D / D               nein                         ja   
1              D / D               nein                         ja   
2              D / D               nein                       nein   
3              D / D               nein                         ja   
4              D / D               nein                         ja   
..               ...                ...                        ...   
109  nicht vorhanden               Nein                       Nein   
110              D/D               nein                       nein   
111              D/D               nein                       nein   
112              D/D               nein                       nein   
113             nein               nein                         ja   

    taster für Schließung__ja_nein vorsicherung__ja_nein sz_magnet__ja_nein  \
0                               ja                  nein               nein   
1                               ja                  nein               nein   
2                             nein                  nein               nein   
3                               ja                  nein               nein   
4                               ja                  nein               nein   
..                             ...                   ...                ...   
109                           Nein                  Nein               Nein   
110                           nein                  nein               nein   
111                           nein                  nein               nein   
112                           nein                  nein               nein   
113                             ja                  nein               nein   

                   Bemerkung  room door integration_aks  
0                im Plan rot  2205   01  420 K1A2205.01  
1                im Plan rot  2506   01  420 K1A2506.01  
2                        NaN  2308   01  420 K1A2308.01  
3                im Plan rot  2108   01  420 K1A2108.01  
4                im Plan rot  1305   02  420 K1A1305.02  
..                       ...   ...  ...             ...  
109                Keine Tür  0226   02  420 02A0226.02  
110  keine Herstellerangaben  0216   01  420 DAA0216.01  
111                      NaN  0213   02  420 DAA0213.02  
112                      NaN  0213   01  420 DAA

## BST File (Done 🆗)

In [2]:
b = BSTLoader(file="data/Sisando_BST_Obj420.xlsx", title="BST")
b.get_data(prefixed=True)

BST___name BST___ebene BST___nummer BST___status_text BST___status  \
0         420          00     A0201.01                OK            1   
1         420          00     A0203.01                OK            1   
2         420          00     A0803.01                OK            1   
3         420          00     A0803.02           HINWEIS           21   
4         420          00     A0807.03                OK            1   
..        ...         ...          ...               ...          ...   
69        420          02     A0617.01                OK            1   
70        420          02     A0618.01                OK            1   
71        420          02     A0714.01                OK            1   
72        420          DA     A0216.01                OK            1   
73        420          DA     A0714.01                OK            1   

      BST___class BST___raum_2_1 BST___raum_2_2  \
0   Drehflügel #1          A0201            NaN   
1   Drehflügel #2          A0203            NaN   
2   Drehflügel #1          A0803            NaN   
3   Drehflügel #2          A0803            NaN   
4   Drehflügel #2          A0807            NaN   
..            ...            ...            ...   
69  Drehflügel #1          A0617            NaN   
70  Drehflügel #1          A0617            NaN   
71  Drehflügel #1          A0714            NaN   
72  Drehflügel #2          A0213            NaN   
73  Schiebetor #1          A0213            NaN   

                             BST___tow_tod__anmerkung  \
0                      STATUS OK_x001D__x001D__x001D_   
1                      STATUS OK_x001D__x001D__x001D_   
2                      STATUS OK_x001D__x001D__x001D_   
3   STATUS OK_x001D_H000 Freilauftürschließer hält...   
4                      STATUS OK_x001D__x001D__x001D_   
..                                                ...   
69                     STATUS OK_x001D__x001D__x001D_   
70                     STATUS OK_x001D__x001D__x001D_   
71                     STATUS OK_x001D__x001D__x001D_   
72                     STATUS OK_x001D__x001D__x001D_   
73                     STATUS OK_x001D__x001D__x001D_   

   BST___tow_tod_last__durchführung BST___tow_tod_last__pnl_name  \
0               2024-10-28 00:00:00                          peD   
1               2024-10-28 00:00:00                          peD   
2               2024-10-28 00:00:00                          peD   
3               2024-10-28 00:00:00                          peD   
4               2024-10-28 00:00:00                          peD   
..                              ...                          ...   
69              2024-10-28 00:00:00                          peD   
70              2024-10-28 00:00:00                          peD   
71              2024-10-28 00:00:00                          peD   
72              2024-10-28 00:00:00                          peD   
73              2024-10-28 00:00:00                          peD   

   BST___pnl_tow_tod_last__f_name BST___e_antrieb BST___exf_01  \
0                   David PESCHEK             NaN          NaN   
1                   David PESCHEK             NaN            1   
2                   David PESCHEK             NaN            1   
3                   David PESCHEK             NaN            1   
4                   David PESCHEK             NaN          NaN   
..                            ...             ...          ...   
69                  David PESCHEK             NaN            1   
70                  David PESCHEK             NaN          NaN   
71                  David PESCHEK             NaN            1   
72                  David PESCHEK             NaN            1   
73                  David PESCHEK             NaN          NaN   

   BST___haltemechanik BST___hersteller BST___klassifikation  \
0                  NaN          Peneder                  T30   
1                  NaN          Peneder             EI2-30-C   
2                  NaN    

## FLT Liste (Done 🆗)

In [3]:
f = FLTLoader(file="data/Sisando_FLT_Obj420.xlsx", title="FLT")
f.get_data(prefixed=True)

FLT___lfd_nr FLT___plan_nr FLT___bauteil FLT___ebene FLT___modul  \
0             1        420_eg             A          EG           A   
1             2        420_eg             A          EG         NaN   
2             3        420_eg             A          EG         NaN   
3             4        420_eg             A          EG         NaN   
4             5        420_eg             A          EG         NaN   
..          ...           ...           ...         ...         ...   
78          NaN        420_o2           420           2         NaN   
79          NaN        420_o2           420           2         NaN   
80          NaN        420_o3           420          DG           A   
81          NaN        420_o3           420          DG         NaN   
82          NaN        420_o3           420          DG         NaN   

   FLT___raum_nr FLT___tuer_nr FLT___fluchtmöglichkeit_ja  \
0           3108            01                          X   
1           5206            05                        NaN   
2           4401            07                          X   
3           2601            06                          X   
4           2601            04                          X   
..           ...           ...                        ...   
78        0A0416     0A0416.02                          X   
79        0A0414     0A0414.01                          X   
80        0A0216     0A0216.01                        NaN   
81        0A0219     0A0213.01                        NaN   
82        0A0202     0A0213.02                          X   

   FLT___fluchtmöglichkeit_nein FLT___blindzylinder_rosette_abdeckung  ...  \
0                           NaN                                   NaN  ...   
1                             X                                   NaN  ...   
2                           NaN                                   NaN  ...   
3                           NaN                                   NaN  ...   
4                           NaN                                   NaN  ...   
..                          ...                                   ...  ...   
78                          NaN                                     X  ...   
79                          NaN                                   NaN  ...   
80                            X                                   NaN  ...   
81                            X                                   NaN  ...   
82                          NaN                                   NaN  ...   

   FLT___nottaster_nein                                  FLT___mängel_art  \
0                     X                                               NaN   
1                     X  Panikschloss fehlt (Außentür, Offline-Beschlag?)   
2                     X                                               NaN   
3                     X                                               NaN   
4                     X                                               NaN   
..                  ...                                               ...   
78                    X                                               NaN   
79                    X                                               NaN   
80                    X                                Panikschloss fehlt   
81                    X                                Panikschloss fehlt   
82                    X                                               NaN   

   FLT___maengel_erledigt FLT___bewertung_ok FLT___dummy_1 FLT___dummy_2  \
0                     NaN                NaN           NaN           NaN   
1                     NaN                NaN           NaN           NaN   
2                     NaN                NaN           NaN           NaN   
3                     NaN                NaN           NaN           NaN   
4                     NaN                NaN           NaN           NaN   
..                    ...                ...           ...           ...   
78                    N

In [3]:
f.get_data(prefixed=True)["merge"]

0                420 00A3108.01
1                420 00A5206.05
2                420 00A4401.07
3                420 00A2601.06
4                420 00A2601.04
                ...            
78    420 024200A0416.0A0416.02
79    420 024200A0414.0A0414.01
80    420 DA4200A0216.0A0216.01
81    420 DA4200A0219.0A0213.01
82    420 DA4200A0202.0A0213.02
Name: merge, Length: 83, dtype: object

## FM Liste (Done 🆗)

In [2]:
f = FMLoader(file="data/BSTüren_Filemaker_11.04.22.xlsx", title="FM")
f.get_data(prefixed=True)

FM___bauteil FM___ebene FM___topnr FM___brandmelderkette  \
4594          420         00         78                   021   
4595          420         00         79                  018A   
4596          420         00         80                   006   
4597          420         00         81                \nBüro   
4598          420         00         82                   NaN   

     FM___brandmeldernr FM___erfasst_von FM___tuer_12flg FM___zugaengigkeit  \
4594           B2605.01            Block             NaN                  1   
4595           B1607.03            Block             NaN                  1   
4596           B1607.01            Block             NaN          2 FLG. Li   
4597           B1605.02            Block             NaN          1 FLG. Re   
4598                012            Block             NaN          1 FLG. Li   

     FM___ok FM___hersteller  ... FM___dummy_23 FM___dummy_24 FM___dummy_25  \
4594     NaN             NaN  ...       6782013           NaN           NaN   
4595     NaN             NaN  ...       6782013           NaN           NaN   
4596     NaN             NaN  ...       6782013           NaN           NaN   
4597     NaN             NaN  ...       6782013           NaN           NaN   
4598     NaN       Kollektor  ...       6782013           NaN           NaN   

     FM___dummy_26                                      FM___dummy_27  \
4594           NaN                                                NaN   
4595           NaN                                                NaN   
4596           NaN  Quellstreifen bei allen alten Türen die neu la...   
4597           NaN                                                NaN   
4598           NaN                                                NaN   

     FM___dummy_28 FM___dummy_29 FM___dummy_30 FM___integration_aks  \
4594           744            10            04       420 00B2605.01   
4595           744            10            04       420 00B1607.03   
4596           744            10            04       420 00B1607.01   
4597           744            10            04       420 00B1605.02   
4598           744            10            04            420 00012   

               merge  
4594  420 00B2605.01  
4595  420 00B1607.03  
4596  420 00B1607.01  
4597  420 00B1605.02  
4598       420 00012  

[5 rows x 162 columns]